pallets/aggregate/src/migrations/v2.rs

```diff
@@ -0,0 +1,214 @@
+use sp_core::Get;
+use sp_std::vec::Vec;
+
+/// Implements [`UncheckedOnRuntimeUpgrade`], migrating the state of this pallet from V0 to V1.
+///
+/// In V0 of the template, the value of the [`crate::Domains`] `StorageMap` is the old `Domain`
+/// without delivery configurations and aggregation rules.
+///
+/// We migrate every domain by add a [`crate::data::AggregateSecurityRules::Untrusted`] rules
+/// and a [`crate::data::DeliveryParams`] with the same domain owner, none destination and zero
+/// price.
+///
+/// If we cannot find any delivery owner, we'll remove domain from the storage.
+pub struct InnerMigrateV1ToV2<T>(core::marker::PhantomData<T>);
+
+impl<T: crate::Config> UncheckedOnRuntimeUpgrade for InnerMigrateV1ToV2<T> {
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
+        // Count the _migrable_entries
+        let counts: u64 = v1::Domains::<T>::iter()
+            .filter_map(|(_, d)| d.owner.account().map(|_| Some(())))
+            .count() as u64;
+        use codec::Encode;
+        Ok(counts.encode().to_vec())
+    }
+
+    /// Migrate the storage from V1 to v2.
+    fn on_runtime_upgrade() -> frame_support::weights::Weight {
+        let old_storage = v1::Domains::<T>::drain().collect::<Vec<_>>();
+        let (reads, mut writes) = (old_storage.len() as u64, old_storage.len() as u64);
+        let converted = old_storage
+            .into_iter()
+            .filter_map(|(domain_id, domain)| {
+                // Only the domain with a delivery owner can be migrated
+                domain
+                    .owner
+                    .account()
+                    .cloned()
+                    .map(|owner_id| (domain_id, domain, owner_id))
+            })
+            .map(|(domain_id, domain, owner_id)| {
+                (
+                    domain_id,
+                    crate::Domain::<T>(crate::data::DomainEntry {
+                        id: domain.id,
+                        owner: domain.owner,
+                        state: domain.state,
+                        next: domain.next,
+                        max_aggregation_size: domain.max_aggregation_size,
+                        should_publish: domain.should_publish,
+                        publish_queue_size: domain.publish_queue_size,
+                        ticket: domain.ticket,
+                        aggregate_rules: crate::data::AggregateSecurityRules::Untrusted,
+                        delivery: crate::data::DeliveryParams::new(
+                            owner_id,
+                            crate::data::Delivery::new(
+                                hp_dispatch::Destination::None,
+                                0_u32.into(),
+                                0_u32.into(), // 0 as owner tip initially
+                            ),
+                        ),
+                    }),
+                )
+            })
```

Comment:
That's just wrong!
The old domain is already correct, just `owner_tip` is missed. So the logic should be to get the old domain and copy all fields like they are in the new domain and just add 0 for the `owner_tip` that it wasn't present before.

Reply:
my bad you are right, I will check this out and copy all values except owner tip that will be 0